packages/@react-aria/combobox/src/useComboBox.ts

```diff
@@ -110,12 +110,14 @@ export function useComboBox<T>(props: AriaComboBoxProps<T>, state: ComboBoxState
   let onKeyDown = (e: KeyboardEvent) => {
     switch (e.key) {
       case 'Enter':
-      case 'Tab':
         // Prevent form submission if menu is open since we may be selecting a option
         if (state.isOpen && e.key === 'Enter') {
           e.preventDefault();
         }
-
+        !state.isOpen && props.allowsCustomValue && props.onCustomValueEnterPressed(state.inputValue);
+        state.commit();
+        break;
+      case 'Tab':
```

Comment:
According to the linked issue, we actually just want to call `props.onKeyDown` if provided (like [so](https://github.com/adobe/react-spectrum/issues/2468#issuecomment-946799968)) rather than creating a dedicated prop for onCustomValue submits.

Reply:
The `props.onKeyDown` doesn't return the value, just the keyboard event. Per the linked issue, _"... check the TextEdit value at the time of enter ..."_, the value needs to be checked.

Reply:
Right, but my interpretation of the submitted issue is that `props.onKeyDown` doesn't work for ComboBox, something which the docs actually indicate should be supported. 
![image](https://user-images.githubusercontent.com/9661127/145276517-d92fae72-938e-40e8-9d89-d32d130f4356.png)

Reply:
Lemme talk with the team though, I feel like we might not want to support onKeyDown possibly and in that case something like your `onCustomValueEnterPressed` would be a good substitute

Reply:
Sounds good

Either way, is easy to do I think, let me know the outcome of the discussion.

Reply:
@sdjustin Talked w/ the team, for now we'll want to just have `props.onKeyDown` properly be chained with the `onKeyDown` useComboBox sets up (see https://github.com/adobe/react-spectrum/issues/2468#issuecomment-946799968). We'll likely revisit this custom value submit later down the road, but in the short term we want to align behavior with what the docs describe.

Reply:
@LFDanLu Sounds good, here's the [PR](https://github.com/adobe/react-spectrum/pull/2674) for that.